src/append_only_data.rs

```diff
@@ -663,6 +663,28 @@ where
     P: Permissions + std::hash::Hash + Clone,
 {
     fn append(&mut self, entries: &[(Vec<u8>, Vec<u8>)], last_entries_index: u64) -> Result<()> {
+        let existing: BTreeSet<Vec<u8>> = self
```

Comment:
`existing` isn't used until line 679 below, so better to create it just immediately before there rather than this early.  (e.g. this would would be wasted effort if we return at line 676).